### DISCUSSION TITLE
表格初始化时，多个排序字段怎么初始化？

### DISCUSSION THREAD

**Discussion by luoxgwb** [layui/layui#1832] [问答]
表格设置有两个排序字段，但是initSort只支持指定一个字段进行排序，怎么保持两个甚至多个表头排序显示状态正确？

**Comment by LSL1618** [comment]
多字段排序建议后端接口处理

**Reply by luoxgwb** [reply]
是后端处理的，但是前端表格显示错误呀，例如姓名和性别列排序，选择姓名排序，再选择性别排序，表格无法两个排序显示出来，只会显示最有一次点击的，我需要姓名和性别都显示正确的排序状态，例如，姓名正序，性别倒叙

![image](https://github.com/layui/layui/assets/21021644/280bc937-deaa-40fa-9eba-1d8d08016846)

这里无法同时显示状态

**Reply by bxjt123** [reply]
后端返回排序好的数据，表格就不需要设置`initSort`了。如果点表头的排序，就是表格自动按你点的排序去展示了

**Reply by luoxgwb** [reply]
看我的问题呀，如果两个排序，点了第一个列排序，重载数据，点第二个列排序的时候，重载数据就会重置排序图标，第一个就变成灰色的了，数据是按照要求排序的，但是表头的图标和排序规则对不上呀

**Reply by bxjt123** [reply]
表头的排序图标不是为了展示后端的排序规则，而是在**仅使用前端排序**功能时所展示的一个排序

**Reply by LSL1618** [reply]
@luoxgwb 有一点你搞错了，table组件的列排序功能只支持单列排序，不支持多列组合排序。不管点击哪一列，都是以当前列为基准进行排序，与其它排序列互相独立，互不干扰。所以你要的多列组合排序还是交给后端接口或者SQL(order by)去处理为好。
